src/parsers/manifest/dash/common/parse_representations.ts

```diff
@@ -213,6 +213,17 @@ export default function parseRepresentations(
       codecs = codecs === "mp4a.40.02" ? "mp4a.40.2" : codecs;
       parsedRepresentation.codecs = codecs;
     }
+
+    let supplementalCodecs: string | undefined;
+    if (representation.attributes.supplementalCodecs != null) {
+      supplementalCodecs = representation.attributes.supplementalCodecs;
+    } else if (adaptation.attributes.supplementalCodecs != null) {
+      supplementalCodecs = adaptation.attributes.supplementalCodecs;
+    }
+    if (supplementalCodecs != null) {
+      parsedRepresentation.supplementalCodecs = supplementalCodecs;
```

Comment:
You can avoid creating this useless `supplementalCodecs` variable and directly set the values to the parsedRepresentation.supplementalCodecs, like this : 

if (representation.attributes.supplementalCodecs != null) {
      parsedRepresentation.supplementalCodecs =  representation.attributes.supplementalCodecs;
} else if (adaptation.attributes.supplementalCodecs != null) {
      parsedRepresentation.supplementalCodecs = adaptation.attributes.supplementalCodecs;
}

Reply:
I changed with your recommandation in a first place.
But I have follow Paul recommandation and I have moved the codec conversion logic that is now done at this place.
I think creating a variable `supplementalCodecs` is now useful as operations are done on this variable.